src/index.ts

```diff
@@ -1,16 +1,177 @@
+			);
+		}
+		return sender;
+	};
+
+	/**
+	 * Places an item for sale on the marketplace using Web3.js.
+	 *
+	 * @param {Contract} contract - The initialized Web3 contract instance of the marketplace.
+	 * @param {string} nftContract - The address of the NFT contract.
+	 * @param {string} tokenId - The ID of the token in the NFT contract.
+	 * @param {string} price - The listing price in wei as a string.
+	 * @param {string} deadline - The listing deadline as a UNIX timestamp in seconds, or "0" if no deadline.
+	 * @returns {Promise<Receipt>} - Returns the transaction receipt object.
+	 */
+	public async listItem(
+		contract: Contract<typeof marketplaceABI>,
+		nftContract: string,
+		tokenId: string,
+		price: string,
+		deadline: string,
+	): Promise<Receipt> {
+		try {
+			const sender = await this.getSender();
+
+			const receipt = await contract.methods.listItem(nftContract, tokenId, price, deadline).send({
+				from: sender,
+			});
+
+			return receipt;
+		} catch (error) {
+			console.error('Error listing item:', error);
+			throw error;
+		}
+	}
+	/**
+	 * Places multiple items for sale on the marketplace using Web3.js.
+	 *
+	 * @param {Object} contract - The initialized Web3 contract instance of the marketplace.
+	 * @param {string[]} nftContracts - Array of NFT contract addresses.
+	 * @param {string[]} tokenIds - Array of token IDs in the NFT contracts.
+	 * @param {string[]} amounts - Array of token amounts as strings.
+	 * @param {string[]} prices - Array of listing prices in wei as strings.
+	 * @param {string[]} deadlines - Array of listing deadlines as UNIX timestamps in seconds.
+	 * @returns {Promise<Receipt>} - Returns the transaction receipt object.
+	 */
+	public async listItems(
+		contract: Contract<typeof marketplaceABI>,
+		nftContracts: string[],
+		tokenIds: string[],
+		amounts: string[],
+		prices: string[],
+		deadlines: string[],
+	): Promise<Receipt> {
+		try {
+			const sender = await this.getSender();
+
+			const receipt = await contract.methods
+				.listItems(nftContracts, tokenIds, amounts, prices, deadlines)
+				.send({
+					from: sender,
+				});
+
+			return receipt;
+		} catch (error) {
+			console.error('Error listing items:', error);
+			throw error;
+		}
+	}
+
+	/**
+	 * Cancels the listing of the token and returns it to the seller using Web3.js.
+	 *
+	 * @param {Contract<typeof marketplaceABI>} contract - The initialized Web3 contract instance of the marketplace.
+	 * @param {string} itemId - The ID of the listed token on the marketplace.
+	 * @returns {Promise<Receipt>} - Returns the transaction receipt object.
+	 */
+	async cancelListing(contract: Contract<typeof marketplaceABI>, itemId: string): Promise<Receipt> {
+		try {
+			const sender = await this.getSender();
+
+			// Send the transaction to call the `cancelListing` function
+			const receipt = await contract.methods.cancelListing(itemId).send({
+				from: sender,
+			});
+
+			console.log('Listing canceled successfully:', receipt);
```

Comment:
Is this supposed to be here? I don't see `console.log` messages for any of the above functions.

Reply:
good catch, removing those